File: 3314-construct-the-minimum-bitwise-array-i/3314-construct-the-minimum-bitwise-array-i.go
func minBitwiseArray(nums []int) []int {
    ans := make([]int, len(nums))
    for i, num := range nums {
        val := -1
        for j := 1; j < num; j++ {
            if (j | (j + 1)) == num {
                val = j
                break
            }
        }
        ans[i] = val
    }

    return ans
}

// array, bit manipulation
// time: O(nm)
// space: O(n)

File: 3314-construct-the-minimum-bitwise-array-i/3314-construct-the-minimum-bitwise-array-i.py
class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        ans = [0 for _ in range(len(nums))]
        for i in range(len(nums)):
            val = -1
            for j in range(1, nums[i]):
                if j | (j + 1) == nums[i]:
                    val = j
                    break
            ans[i] = val
        
        return ans

# aray, bit manipulation
# time: O(nm)
# space: O(n)